tests/test_url_build.py

```diff
@@ -32,12 +32,21 @@ def test_build_with_scheme_and_host():
     assert u == URL("http://127.0.0.1")
 
 
-def test_build_with_port():
-    with pytest.raises(ValueError):
-        URL.build(port=8000)
-
-    u = URL.build(scheme="http", host="127.0.0.1", port=8000)
-    assert str(u) == "http://127.0.0.1:8000"
+@pytest.mark.parametrize(
+    ("port", "exc", "match"),
+    [
+        pytest.param(
+            8000,
+            ValueError,
+            r'^Can\'t build URL with "port" but without "host".$',
+            id="port-only",
+        ),
+        pytest.param("", TypeError, r"^port is required to be int$", id="port-str"),
```

Comment:
Looking at the other exception messages, I realized that this one doesn't match the common style. Let's change it to look similar. For example, all the other messages are full sentences, starting with a capital letter and ending with a period.

Reply:
The common pattern does not end with a period.
```
# Ends with period
grep raise yarl/ -R | grep -E "\..\)" | wc -l
# 2
```

```
# Ends with anything but period
grep raise yarl/ -R | grep -E "[^\.].\)$" | wc -l
58
```